client/search/search-input.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import SearchIcon from '../icons/material/baseline-search-24px.svg'
+import { useKeyListener } from '../keyboard/key-listener'
+import { fastOutSlowInShort } from '../material/curves'
+import TextField from '../material/text-field'
+import { useValueAsRef } from '../state-hooks'
+
+const ESCAPE = 'Escape'
+const F = 'KeyF'
+
+const TextFieldContainer = styled(TextField)`
+  width: ${props => (props.isFocused ? '250px' : '200px')};
+  ${fastOutSlowInShort};
+`
+
+interface SearchInputProps {
+  searchQuery: string
+  onSearchChange: (value: string) => void
+  className?: string
+}
+
+export function SearchInput({ searchQuery, onSearchChange, className }: SearchInputProps) {
+  const [inputValue, setInputValue] = useState('')
```

Comment:
this should be initialized to `searchQuery`, otherwise you're needlessly double-rendering every time this component starts with a query